### PR TITLE
feat(TableSettings): add options for compatibility with other features

### PR DESCRIPTION
**Pull request by TaPo4eK111291**

### CODE REVIEW

File: src/constants/selectionColumn.tsx [hunk 1]
```diff
@@ -22,6 +22,9 @@ export const selectionColumn: ColumnDef<unknown> = {
             onChange={row.getToggleSelectedHandler()}
         />
     ),
+    meta: {
+        hideInSettings: true,
```
Comment (beliarh) [inline src/constants/selectionColumn.tsx:26]: Add the same meta in `actionsColumn` and `dragHandleColumn`

File: src/components/TableSettings/__stories__/stories/TableSettingsColumnStory.tsx [hunk 2]
```diff
@@ -97,17 +97,21 @@ const columns: ColumnDef<Item>[] = [
...
+    const [rowSelection, setRowSelection] = React.useState<RowSelectionState>({});
     const settingsColumn = getSettingsColumn<Item>(SETTINGS_COLUMN_ID, options);
     const [columnPinning, setColumnPinning] = React.useState<ColumnPinningState>({
-        left: [],
+        left: ['_select'],
         right: [SETTINGS_COLUMN_ID],
     });
     const table = useTable({
-        columns: [...columns, settingsColumn],
+        columns: [selectionColumn as ColumnDef<Item>, ...columns, settingsColumn],
```
Comment (beliarh) [inline src/components/TableSettings/__stories__/stories/TableSettingsColumnStory.tsx:107]: Move `columns` value to `useMemo`

File: src/components/TableSettings/TableSettings.extentions.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import type {Column, RowData} from '@tanstack/react-table';
+
+declare module '@tanstack/react-table' {
+    interface ColumnMeta<TData extends RowData, TValue> {
+        hideInSettings?: boolean;
+        inSettings?: ((column: Column<TData, TValue>) => string | React.ReactNode) | string;
```
Comment (beliarh) [inline src/components/TableSettings/TableSettings.extentions.ts:6]: 1. Maybe `titleInSettings`?
2. `((column: Column<TData, TValue>) => React.ReactNode) | React.ReactNode` (`ReactNode` already includes `string`)

File: README.md [hunk 4]
```diff
@@ -359,8 +359,12 @@ const columns: ColumnDef<Person>[] = [
   // ...other columns
   {
     id: 'settings_column_id',
-    header: (context) => <TableSettings {...context} columnId="settings_column_id" />,
-  }, // or u can use function getSettingsColumn
+    header: ({table}) => <TableSettings table={table} />,
+    meta: {
+      hideInSettings: false, // Optional. Allows to hide this column from settings popover
+      inSettings: 'some string or any html', // Optional. Overrides header field for settings popover (if you need different content for header and settings popover)
```
Comment (beliarh) [inline README.md:365]: > `any html`

Do you mean ReactNode? Because html is escaped by react

Comment (TaPo4eK111291) [reply README.md]: changed to ReactNode

File: src/components/TableSettings/TableSettings.extentions.ts (hunk 3 above)
Comment (TaPo4eK111291) [reply src/components/TableSettings/TableSettings.extentions.ts]: Fixed both

File: src/components/TableSettings/__stories__/stories/TableSettingsColumnStory.tsx (hunk 2 above)
Comment (TaPo4eK111291) [reply src/components/TableSettings/__stories__/stories/TableSettingsColumnStory.tsx]: Moved to useMemo

File: src/constants/selectionColumn.tsx (hunk 1 above)
Comment (TaPo4eK111291) [reply src/constants/selectionColumn.tsx:26]: Added

File: src/components/TableSettings/__stories__/stories/TableSettingsColumnStory.tsx [hunk 5]
```diff
@@ -97,17 +97,27 @@ const columns: ColumnDef<Item>[] = [
...
     const settingsColumn = getSettingsColumn<Item>(SETTINGS_COLUMN_ID, options);
     const [columnPinning, setColumnPinning] = React.useState<ColumnPinningState>({
-        left: [],
+        left: ['_select'],
         right: [SETTINGS_COLUMN_ID],
     });
+
+    const columns = React.useMemo(
+        () => [selectionColumn as ColumnDef<Item>, ...tableColumns, settingsColumn],
+        [settingsColumn],
```
Comment (beliarh) [inline src/components/TableSettings/__stories__/stories/TableSettingsColumnStory.tsx:109]: Move `getSettingsColumn` to this `useMemo`:
```
export const TableSettingsColumnStory = ({sortable, filterable}: TableSettingsOptions) => {
    // ...

    const columns = React.useMemo(
        () => [
            selectionColumn as ColumnDef<Item>,
            ...tableColumns,
            getSettingsColumn<Item>(SETTINGS_COLUMN_ID, {sortable, filterable}),
        ],
        [sortable, filterable],
    );

    // ...
}
```

**Review by beliarh — CHANGES_REQUESTED** [review]

File: src/components/TableSettings/__stories__/stories/TableSettingsColumnStory.tsx (hunk 5 above)
Comment (TaPo4eK111291) [reply src/components/TableSettings/__stories__/stories/TableSettingsColumnStory.tsx]: Moved

**Review by beliarh — APPROVED** [review]

**Pull request merged**